flux-typeck/src/type_env/paths_tree.rs

```diff
@@ -189,6 +204,7 @@ impl PathsTree {
                     return match self.lookup_place_iter(rcx, gen, ptr_path.clone(), proj) {
                         LookupResult::Ptr(_, ty2) => LookupResult::Ref(rk, ty2),
                         LookupResult::Ref(rk2, ty2) => LookupResult::Ref(rk.min(rk2), ty2),
+                        LookupResult::Box(ty2) => LookupResult::Box(ty2),
```

Comment:
We should return `LookupResult::Ref(rk, ty2)` here. This code is exercised when you have `x: &Box<T>` and then do `**x`, the result is of type `T` but it's behind a shared reference so we keep track of it.

Reply:
ok!